projects/vdk-core/src/vdk/api/plugin/plugin_input.py

```diff
@@ -294,15 +299,20 @@ def ingest_payload(self,
 
                 return metadata
 
-        :exceptions: Exceptions raised while ingesting data can either be
-        handled at plugin level, or they will be automatically caught at the
-        vdk-core level.
+        :exception: Exceptions raised while ingesting data can either be
+            handled at plugin level, or they will be automatically caught at
+            the vdk-core level.
         """
         pass
 
     def pre_ingest_process(
-        self, payload: List[dict], metadata: Optional[IngestionMetadata] = None
-    ) -> (List[dict], Optional[IngestionMetadata]):
+        self,
+        payload: List[dict],
+        destination_table: Optional[str] = None,
+        target: Optional[str] = None,
+        collection_id: Optional[str] = None,
+        metadata: Optional[IngestionMetadata] = None,
+    ) -> Union[List[Dict], Tuple[List[Dict], IngestionMetadata]]:
```

Comment:
The reason for the change is that it is possible that a user wants to do pre-processing, and return the payload, but not metadata. In such case, the new return types allow for either the payload to be returned, or a tuple of payload and metadata. With the old return signature, plugin developers would have needed to return something like `(payload, )` to avoid type errors.

Reply:
Would you share an example code snippet for usages before/after the return types change?

Also, have you considered always returning the metadata, either modified or directly the parameter value provisioned?

Reply:
To answer your first question, sure, here's an example,
Before:
```python
def pre_ingest_process(self,
    payload: List[dict],
    destination_table: Optional[str],
    target: Optional[str],
    collection_id: Optional[str],
    metadata: Optional[IngestionMetadata]
) -> (List[dict], Optional[IngestionMetadata]):
    for i in payload:
        i["additional_column"] = "some_val"
    return (payload, None)
```

Now:
```python
def pre_ingest_process(self,
    payload: List[dict],
    destination_table: Optional[str],
    target: Optional[str],
    collection_id: Optional[str],
    metadata: Optional[IngestionMetadata]
) -> Union[List[Dict], Tuple[List[Dict], IngestionMetadata]]:
    for i in payload:
        i["additional_column"] = "some_val"
    return payload
```

Reply:
As for your second question, you mean to force plugin developers to always return metadata?

Reply:
on the first question, I see; 
should a plugin developer return ingestion metadata always? if it's a part of the contact, and it is always being the output of an operation that always becomes the input of the next one in the sequence, it may be more clear what exactly becomes the input of the "pre2" for example?

My point is, the API should be the simplest and the most obvious way to understand the workflow.
When having a sequence, if I miss to return the second parameter, does that mean the next pre-processor will be provisioned with None metadata, or else what other metadata?
When always returning the metadata, it becomes quite obvious what the next pre-processor is going to be provisioned with.

Reply:
@ivakoleva 
Seems fair. Ok, I've refactored the method signature, and the docstring to be more explicit as to what the contract is.

Reply:
```
metadata: Optional[IngestionMetadata] = None,
    ) -> Tuple[List[Dict], IngestionMetadata]:

```
the method param is optional, yet the return type is not;
would rather keep them consistent;

Reply:
Yup, fix it, thanks